scio/src/main/scala/org/apache/zeppelin/scio/package.scala

```diff
@@ -0,0 +1,185 @@
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from KV SCollection. */
+    def closeAndDisplay(): Unit = {
+      val it = materialize(self).waitForResult().value
+
+      if (it.isEmpty) {
+        println(SCollectionEmptyMsg)
+      } else {
+        val content = it.take(maxResults).map{ case (k, v) => s"$k$tab$v" }.mkString(newline)
+        println(s"$table key${tab}value$newline$content")
+        notifIfTruncated(it)
+      }
+    }
+
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinProductSCollection[T: ClassTag](val self: SCollection[T])(implicit ev: T <:< Product) extends AnyVal {
+  implicit class ZeppelinProductSCollection[T: ClassTag](val self: SCollection[T])
+                                                        (implicit ev: T <:< Product) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from Product like SCollection */
+    def closeAndDisplay(): Unit = {
+      val it = materialize(self).waitForResult().value
+
+      if (it.isEmpty) {
+        println(SCollectionEmptyMsg)
+      } else {
+        val first = it.next()
+        //TODO is this safe field name to value iterator?
+        val fieldNames = first.getClass.getDeclaredFields.map(_.getName)
+
+        val header = fieldNames.mkString(tab)
+        val firstStr = first.productIterator.mkString(tab)
+        val content = it.take(maxResults).map(_.productIterator.mkString(tab)).mkString(newline)
+        println(s"$table $header$newline$firstStr$newline$content")
+        notifIfTruncated(it)
+      }
+    }
+  }
+
+  // TODO: scala 2.11
+  // implicit class ZeppelinAvroSCollection[T: ClassTag](val self: SCollection[T])(implicit ev: T <:< GenericRecord) extends AnyVal {
+  implicit class ZeppelinAvroSCollection[T: ClassTag](val self: SCollection[T])
+                                                     (implicit ev: T <:< GenericRecord) {
+    /** Convenience method to close the current [[com.spotify.scio.ScioContext]]
+     * and display elements from Avro like SCollection */
+    def closeAndDisplay(schema: Schema = null): Unit = {
+      val it = materialize(self).waitForResult().value
+
+      if (it.isEmpty) {
+        println(SCollectionEmptyMsg)
+      } else {
+        val first = it.next()
+        import collection.JavaConverters._
+
+        val fieldNames = if (schema != null) {
+          schema.getFields.iterator().asScala.map(_.name()).toArray
+        } else {
+          first.getSchema.getFields.iterator.asScala.map(_.name()).toArray
+        }
+
+        val header = fieldNames.mkString(tab)
+        val firstStr = fieldNames.map(first.get).mkString(tab)
+        val content = it.take(maxResults)
+                        .map(r => fieldNames.map(r.get).mkString(tab))
+                        .mkString(newline)
+        println(s"$table $header$newline$firstStr$newline$content")
```

Comment:
What is the purpose of println ? Is it for logging or display ? If it is for display, then does it support table format of zeppelin ?